### ISSUE TITLE
Decide how to format descriptions for complex ranges

### ISSUE THREAD

**Issue by kimrutherford** [pombase/canto#1079]
Ranges like "FYPO_EXT:0000004|%" and "TranscriptID|SO:0000673" are still handled badly.  We need to decide how to make them more obvious.  Here's what penetrance looks like:

![screenshot 2016-02-18 23 03 06](https://cloud.githubusercontent.com/assets/90474/13139920/edce82a6-d693-11e5-8462-d3cc202cd357.png)

**Comment by mah11** [comment]
These seem to be working OK for those of us who already know what it means, so I think the outstanding issue is to make the interface less confusing.

One thing that should help is to move the display text up above the options, so it doesn't have to be repeated. A couple of examples (fugly b/c text; I can do prettier mockups if that'll help):

    penetrance
    [x] [choose qualitative level]
    OR
    [ ] enter quantitative value as percent (e.g. 45%) [___]

    assayed transcript
    [x] type a description to find a SO term [___]
    OR
    [ ] [choose a gene] add ..

Too wordy?

**Comment by ValWood** [comment]
I think its clear, and not so wordy....

Shall we add Mouse over  'information' help text ? for 'penetrance':
( even some geneticists seem remarkably unfamiliar with the terms penetrance or expressivity, and sometimes confuse them)

"Proportion of  the population expressing a phenotype"

**Comment by mah11** [comment]
yep, sounds useful

**Comment by kimrutherford** [comment]
I think this sounds good.  My only concern is that it might be a bit wide for the pop-up box.  Especially this one:

    [ ] enter quantitative value as percent (e.g. 45%) [___]

We may need to trim the text a bit.  I'll try it and send a screenshot.


> enter quantitative value as percent (e.g. 45%) [___]

Currently the "%" ranges are just text fields so I need to support them properly so.  I'll make a separate issue for that.

For SO should the text be the same if the range can only be a SO term?

For example for this one:

    FYPO:0002576    is_a    assayed_using   SO:0001411      assayed region  0,1     user

should it display?:

    assayed region
    type a description to find a SO term [___]

**Comment by ValWood** [comment]
we could just say :
quantitative value  (e.g. 45%)

I thought for SO we had different descriptions depending on the term?

**Comment by kimrutherford** [comment]
> quantitative value (e.g. 45%)

That would fit the popup box better.

> I thought for SO we had different descriptions depending on the term?

We put the term name in the autocomplete box, but it doesn't work very well because some of the term names are a lot longer than the box.

**Comment by mah11** [comment]
>> quantitative value (e.g. 45%)
>
> That would fit the popup box better.

fine by me

For SO terms, I'm not sure I understand the question(s) correctly, but I'll tell you what I think anyway ;)

This should be fine as a general approach:

    [display text from config file]
    type a description to find a SO term [___]

e.g. for SO only

    assayed region
    type a description to find a SO term [___]

or

    assayed transcript
    type a description to find a SO term [___]

and if SO or gene, as above:

    assayed transcript
    [x] type a description to find a SO term [___]
    OR
    [ ] [choose a gene] add ..

I don't really mind whether there's a greyed-out suggestion in the autocomplete box.

**Comment by ValWood** [comment]
I'll go with that. about to do testing

**Comment by kimrutherford** [comment]
> Shall we add Mouse over 'information' help text ? for 'penetrance':

Good plan.  We'll need an extra column in the configuration file for that.

**Comment by kimrutherford** [comment]
> This should be fine as a general approach:

Thanks for the ascii diagrams.  That helped a lot.  I think it now more or less looks like you've described.  Probably good if we have a Skype chat soon to go over it and discuss fine tuning.

**Comment by mah11** [comment]
I'm happy to chat, but I think it looks fine and works as expected now.
